### PR TITLE
Update Python Dependencies

### PR DESCRIPTION
**Pull request by brandonlenz**
## 🗣 Description ##

Resolves: #666 (😨)

Closes:
- https://github.com/cisagov/getgov/pull/644
- https://github.com/cisagov/getgov/pull/643
- https://github.com/cisagov/getgov/pull/616
- https://github.com/cisagov/getgov/pull/615

Ran the script outlined in https://github.com/cisagov/getgov/blob/main/docs/operations/runbooks/update_python_dependencies.md

## 💭 Motivation and context ##

There are some open dependabot PRs that update requirements.txt but not the pipfile.lock
To keep the two in sync, we have manually generate a new Pipfile.lock and requirements.txt

Note: This updates ALL dependencies, not just the ones we're interested in, since dependencies are pinned to `*` in the Pipfile.

This includes some major version changes, which causes alarm if the dependency follows semver and includes breaking changes. Smoke testing is encouraged, alongside automated test results.

### CODE REVIEW

File: src/requirements.txt [hunk 1]
```diff
@@ -29,21 +29,22 @@ jmespath==1.0.1 ; python_version >= '3.7'
...
-pycryptodomex==3.17
+pycryptodomex==3.18.0
+pydantic==1.10.8 ; python_version >= '3.7'
 pyjwkest==1.4.2
 python-dateutil==2.8.2 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
-python-dotenv==0.21.1 ; python_version >= '3.7'
-requests==2.28.2
-s3transfer==0.6.0 ; python_version >= '3.7'
-setuptools==67.2.0 ; python_version >= '3.7'
+python-dotenv==1.0.0 ; python_version >= '3.8'
```
Comment (brandonlenz) [inline src/requirements.txt:42]: Major version bump

File: src/requirements.txt [hunk 2]
```diff
@@ -29,21 +29,22 @@ jmespath==1.0.1 ; python_version >= '3.7'
...
+requests==2.31.0
+s3transfer==0.6.1 ; python_version >= '3.7'
+setuptools==67.8.0 ; python_version >= '3.7'
 six==1.16.0 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
-sqlparse==0.4.3 ; python_version >= '3.5'
-typing-extensions==4.4.0 ; python_version >= '3.7'
-urllib3==1.26.14 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5'
-whitenoise==6.3.0
+sqlparse==0.4.4 ; python_version >= '3.5'
+typing-extensions==4.6.2 ; python_version < '3.11'
```
Comment (brandonlenz) [inline src/requirements.txt:48]: python_version polarity flipped to a less than 🧐

File: src/requirements.txt [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+django-auditlog==2.3.0
 django-cache-url==3.4.4
 django-csp==3.7
 django-fsm==2.8.1
-django-phonenumber-field[phonenumberslite]==7.0.2
+django-phonenumber-field[phonenumberslite]==7.1.0
 django-widget-tweaks==1.4.12
 environs[django]==9.5.0
-faker==17.0.0
+faker==18.10.0
```
Comment (brandonlenz) [inline src/requirements.txt:23]: Major version bump

File: src/requirements.txt [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
-asgiref==3.6.0 ; python_version >= '3.7'
-boto3==1.26.69
-botocore==1.29.69 ; python_version >= '3.7'
-cachetools==5.3.0
-certifi==2022.12.7 ; python_version >= '3.6'
+asgiref==3.7.2 ; python_version >= '3.7'
+boto3==1.26.144
+botocore==1.29.144 ; python_version >= '3.7'
+cachetools==5.3.1
+certifi==2023.5.7 ; python_version >= '3.6'
```
Comment (brandonlenz) [inline src/requirements.txt:6]: Major version bump

File: src/requirements.txt [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
 cfenv==0.5.3
 cffi==1.15.1
-charset-normalizer==3.0.1 ; python_version >= '3.6'
-cryptography==39.0.1 ; python_version >= '3.6'
+charset-normalizer==3.1.0 ; python_full_version >= '3.7.0'
+cryptography==41.0.1 ; python_version >= '3.7'
```
Comment (brandonlenz) [inline src/requirements.txt:10]: Major version bump

File: src/Pipfile [hunk 6]
```diff
@@ -23,6 +23,7 @@ requests = "*"
 django-fsm = "*"
 django-phonenumber-field = {extras = ["phonenumberslite"], version = "*"}
 boto3 = "*"
+typing-extensions ='*'
```
Comment (brandonlenz) [inline src/Pipfile:26]: I was running into an [error where this was not available in CI](https://github.com/cisagov/getgov/actions/runs/5146963106/jobs/9266729233?pr=667). Not sure why it is now explicitly required

Any insight appreciated 😄

**Comment by kewlguy781** [comment]
Have the changelog of respective libraries, (ex, in this case, Pythons) been reviewed for any deprecated functions? (Thats something I would like to see when updating the big libraries, mainly when it come to Python and Django)

**Comment by brandonlenz** [comment]
I'm going to recreate this PR since I borked the branch naming. I want this deployed to sandbox for smoke testing as well. 

> Have the changelog of respective libraries, (ex, in this case, Pythons) been reviewed for any deprecated functions? (Thats something I would like to see when updating the big libraries, mainly when it come to Python and Django)

@kewlguy781 On that PR I'll link some changelogs and my analysis as I'm going through it :)

**Comment by neilmb** [comment]
Cryptography's release notes has nothing marked `BACKWARDS INCOMPATIBLE` that should have any impact on us: https://groups.google.com/g/comp.lang.python.announce/c/Q-iVWGsK0SU?pli=1

**Comment by neilmb** [comment]
Everything else looks like patch and minor release changes.

**Pull request closed without merging**